cli/kata-check_amd64.go

```diff
@@ -155,7 +198,53 @@ func kvmIsUsable() error {
 	return genericKvmIsUsable()
 }
 
-func archHostCanCreateVMContainer() error {
+// acrnIsUsable determines if it will be possible to create a full virtual machine
+// by creating a minimal VM and then deleting it.
+func acrnIsUsable() error {
+	flags := syscall.O_RDWR | syscall.O_CLOEXEC
+
+	f, err := syscall.Open(acrnDevice, flags, 0)
+	if err != nil {
+		return err
+	}
+	defer syscall.Close(f)
+
+	fieldLogger := kataLog.WithField("check-type", "full")
+
+	fieldLogger.WithField("device", acrnDevice).Info("device available")
+
+	createVM := acrn_create_vm{
+		uuid: [16]uint8{
+			0xd2, 0x79, 0x54, 0x38, 0x25, 0xd6, 0x11, 0xe8,
+			0x86, 0x4e, 0xcb, 0x7a, 0x18, 0xb3, 0x46, 0x43,
+		},
+	}
+
+	vm, _, errno := syscall.Syscall(syscall.SYS_IOCTL,
+		uintptr(f),
+		uintptr(ioctl_ACRN_CREATE_VM),
+		uintptr(unsafe.Pointer(&createVM)))
+	if errno != 0 {
+		if errno == syscall.EBUSY {
+			fieldLogger.WithField("reason", "another hypervisor running").Error("cannot create VM")
+		}
+		fieldLogger.Infof("ACRN IOCTL Errno=%d", errno)
+		fieldLogger.WithError(errno).Error("ACRN IOCTL Errno")
+		return errno
+	}
+	defer syscall.Close(int(vm))
+
+	fieldLogger.WithField("feature", "create-vm").Info("feature available")
+
+	return nil
+}
+
+func archHostCanCreateVMContainer(onVMM bool) error {
+
+	if onVMM {
+		return acrnIsUsable()
```

Comment:
This would effectively invalidate kvm check when vmVMM is true. Please do not assume acrn is the only hypervisor we support inside a guest.

Reply:
I see 2 options here,
Option 1: If onVMM is true, first check if ACRN is supported. If that fails, try KVM.
Option 2: Open the configuration.toml and check the hypervisor currently being used. Based on the hypervisor, check if the VM container can be created. 

Let me know which is preferred. If you have any other suggestions, please let me know.

Reply:
Another options is to always check if kvm is usable. And if that fails, check if we are on vmm and if acrn is usable.

Since it becomes hypervisor specific, your option 2 might be better. @jodh-intel wdyt?

Reply:
Hi @jodh-intel any thoughts on the above approach?

Reply:
> I see 2 options here,
> Option 1: If onVMM is true, first check if ACRN is supported. If that fails, try KVM.
> Option 2: Open the configuration.toml and check the hypervisor currently being used. Based on the hypervisor, check if the VM container can be created.
> 
> Let me know which is preferred. If you have any other suggestions, please let me know.

Hi @jodh-intel I was referring to this. I have implemented option#2 for now as it seemed generic. But please let me know if you have something else in mind.